apps/server/src/modules/task/uc/submission.uc.spec.ts

```diff
@@ -0,0 +1,189 @@
+	});
+
+	describe('findAllByTask is called', () => {
+		const createParams = () => {
+			const user = userFactory.buildWithId();
+			const task = taskFactory.buildWithId();
+
+			const submission1 = submissionFactory.buildWithId();
+			const submission2 = submissionFactory.buildWithId();
+			const submissions = [submission1, submission2];
+			const countedSubmissions: Counted<Submission[]> = [submissions, 2];
+
+			return { user, task, submissions, countedSubmissions };
+		};
+
+		describe('WHEN service returns successfully and user is authorized for all submissions', () => {
+			const setup = () => {
+				const { user, task, submissions, countedSubmissions } = createParams();
+
+				submissionService.findAllByTask.mockResolvedValueOnce(countedSubmissions);
+				authorizationService.getUserWithPermissions.mockResolvedValueOnce(user);
+				authorizationService.hasPermission.mockReturnValue(true).mockReturnValue(true);
+
+				return { taskId: task.id, user, submissions, countedSubmissions };
+			};
+
+			it('should call findAllByTask', async () => {
+				const { user, taskId } = setup();
+
+				await submissionUc.findAllByTask(user.id, taskId);
+
+				expect(submissionService.findAllByTask).toHaveBeenCalledWith(taskId);
+			});
+
+			it('should call getUserWithPermissions', async () => {
+				const { user, taskId } = setup();
+
+				await submissionUc.findAllByTask(user.id, taskId);
+
+				expect(authorizationService.getUserWithPermissions).toHaveBeenCalledWith(user.id);
+			});
+
+			it('should call hasPermission', async () => {
+				const { user, taskId, submissions } = setup();
+				const permissionContext = PermissionContextBuilder.read([Permission.SUBMISSIONS_VIEW]);
+
+				await submissionUc.findAllByTask(user.id, taskId);
+
+				expect(authorizationService.hasPermission).toHaveBeenNthCalledWith(1, user, submissions[0], permissionContext);
+				expect(authorizationService.hasPermission).toHaveBeenNthCalledWith(2, user, submissions[1], permissionContext);
+			});
+
+			it('should return submissions', async () => {
+				const { user, taskId, submissions } = setup();
+
+				const result = await submissionUc.findAllByTask(user.id, taskId);
+
+				expect(result).toEqual(submissions);
+			});
+		});
+
+		describe('WHEN service returns successfully and user is authorized for second submission only', () => {
+			const setup = () => {
+				const { user, task, submissions, countedSubmissions } = createParams();
+
+				submissionService.findAllByTask.mockResolvedValueOnce(countedSubmissions);
+				authorizationService.getUserWithPermissions.mockResolvedValueOnce(user);
+				authorizationService.hasPermission.mockReturnValueOnce(false).mockReturnValueOnce(true);
+
+				return { taskId: task.id, user, submissions };
+			};
+
+			it('should return only submission2', async () => {
+				const { user, taskId, submissions } = setup();
+
+				const result = await submissionUc.findAllByTask(user.id, taskId);
+				const expectedResult = [submissions[1]];
```

Comment:
I don't think this line is in the correct place here. Considering the Arrange-Act-Assert pattern this would be in the act phase here. That seems incorrect. Probably it should be in the arrange phase, so in setup or directly under the setup call. But you could also leave out the assignment and just say `expect(result).toEqual([submissions[1]]);`.